src/main/java/org/openrewrite/kotlin/internal/KotlinParserVisitor.java

```diff
@@ -931,19 +931,36 @@ public J visitSuperReference(FirSuperReference superReference, ExecutionContext
     @Override
     public J visitFunctionCall(FirFunctionCall functionCall, ExecutionContext ctx) {
         FirFunctionCallOrigin origin = functionCall.getOrigin();
+        if (origin != FirFunctionCallOrigin.Operator || functionCall instanceof FirImplicitInvokeCall) {
+            return mapFunctionCall(functionCall, origin == FirFunctionCallOrigin.Infix);
+        }
+
+        KtSourceElement source = functionCall.getSource();
+        if (source == null) {
+            throw new UnsupportedOperationException("The source of a FirFunctionCall is null");
+        }
 
         J j;
-        if (origin == FirFunctionCallOrigin.Operator && !(functionCall instanceof FirImplicitInvokeCall)) {
+        String type = Objects.requireNonNull(source.getElementType()).toString();
+        if ("BINARY_EXPRESSION".equals(type) ||
```

Comment:
I'd like to discuss this with @sambsnyd tomorrow before we move forward with this PR. So that we can discuss the pros and cons of the current approach and how to represent the syntax sugar of operator overloading.

My main concern is that we'll already be doing a major version bump due to significant changes and it may be best to do them all at once.

We may do something similar here to handle operator overloads by wrapping J trees in a K tree that allow data flow to work and still parse and print the source correctly.

That would involve detecting the types like BINARY_EXPRESSION and PREFIX_EXPRESSION and wrapping a K tree in the method invocations. However, I am not certain how complicated it will be. So, I'd like to discuss the pros and cons first.